backend-api/src/functions/adapters/dynamoDbAdapter.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  ConditionalCheckFailedException,
+  DynamoDBClient,
+  PutItemCommand,
+  PutItemCommandInput,
+  QueryCommand,
+  QueryCommandInput,
+  QueryCommandOutput,
+} from "@aws-sdk/client-dynamodb";
+import { CreateSessionAttributes } from "../services/sessionService/sessionService";
+import { NodeHttpHandler } from "@smithy/node-http-handler";
+import { NativeAttributeValue, unmarshall } from "@aws-sdk/util-dynamodb";
+
+const sessionStates = {
+  ASYNC_AUTH_SESSION_CREATED: "ASYNC_AUTH_SESSION_CREATED",
+};
+
+export type DatabaseRecord = Record<string, NativeAttributeValue>;
+
+export class DynamoDbAdapter {
+  private readonly tableName: string;
+  private readonly dynamoDbClient = new DynamoDBClient({
+    region: process.env.REGION,
+    maxAttempts: 2,
+    requestHandler: new NodeHttpHandler({
+      connectionTimeout: 29000,
```

Comment:
Should these be changed to the max timeout of the lambda that uses them? Or do we not care? I personally don't really have much thought on it, I just bring it up as we changed these when we were updating lambda time outs recently

Reply:
Yes, I completely forgot about this one as I merged a PR the other day to change all timeouts to match the lambda's. Thank you!